intake/gui/source_select.py

```diff
@@ -0,0 +1,154 @@
+#-----------------------------------------------------------------------------
+# Copyright (c) 2012 - 2019, Anaconda, Inc. and Intake contributors
+# All rights reserved.
+#
+# The full license is in the LICENSE file, distributed with this software.
+#-----------------------------------------------------------------------------
+
+from collections import OrderedDict
+
+import intake
+import panel as pn
+
+from .base import Base
+
+class BaseSelector(Base):
+    preprocess = None
+    options = None
+
+    def callback(self, *events):
+        print(self.panel.name, events)
+        for event in events:
+            if event.name == 'value' and event.new != self.selected:
+                self.select(event.new)
+
+    def update_selected(self):
+        if self.widget.value != self.selected:
```

Comment:
Is the condition here necessary? I'd expect this not to trigger an event if the value hasn't changed. Not saying to remove the check, just checking you didn't add it to get around a param bug.

Reply:
Also it seems like these ``update_X`` methods could be implemented by making the ``options`` and ``selected`` parameters of the class and using ``param.depends('selected', watch=True)``. I understand that you are avoiding these param patterns though, so maybe just ignore me or leave it as a suggestion for the future.

Reply:
Thanks for keeping an eye on this @philippjfr . The choice of the correct pattern here is interesting, to discuss how the param model fits into a GUI-app model. Would be happy to see alternate patterns, but that would likely mean discarding some work later on.

Reply:
yeah I did this one first and I think it is more convoluted then it needs to be. I was trying to make sure that a widget or a call could set the value on the class. But it might be overkill